auth.go

```diff
@@ -0,0 +1,99 @@
+package winrm
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/masterzen/winrm/soap"
+	"launchpad.net/gwacl/fork/http"
+	"launchpad.net/gwacl/fork/tls"
+)
+
+func newCustomTransport(endpoint *Endpoint) (*http.Transport, error) {
+	var (
+		cert tls.Certificate
+		err  error
+	)
+
+	// test if it's valid key,cert client base
+	if isCertAndKey(endpoint.Cert, endpoint.Key) {
+		cert, err = tls.X509KeyPair(endpoint.Cert, endpoint.Key)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// make the custom transport
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: endpoint.Insecure,
+			Certificates:       []tls.Certificate{cert},
+		},
+		//ResponseHeaderTimeout: endpoint.Timeout,
```

Comment:
On the gwacl version there were no ResponseHeaderTimeout attribute. But I will change to the azure fork.